src/Message/CompleteAuthoriseAndCaptureResponse.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Omnipay\IcepayPayments\Message;
+
+/**
+ * The response after complete authorise and capture request.
+ * For this response, we explicitly check what the status is of the payment transaction at Icepay.
+ *
+ * @see http://docs2.icepay.com/payment-process/transaction-status-flow/transaction-statuses/
+ */
+class CompleteAuthoriseAndCaptureResponse extends AbstractResponse
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function isSuccessful(): bool
+    {
+        return isset($this->data['status']) && in_array($this->data['status'], [
+            self::RESPONSE_STATUS_COMPLETED,
+            self::RESPONSE_STATUS_SETTLED,
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * In case there is no status 'cancelled' available yet, look up for a statusCode querystring in the HttpRequest.
```

Comment:
The statusCode in the queryString does exist. If you press the cancel button in the PSP, you will be redirected back to your merchant. While being redirected, you'll see the following (example) queryString.

`?contractProfileId=XXXXXXXXXXX&transactionId=XXXXXXXXXXX&reference=XXXXXXXXXXX&statusCode=CANCELLED&statusDetails=&paymentMethod=CREDITCARD&issuer=MASTER&amountInCents=10500&currencyCode=EUR&providerTransactionId=1234567&checksum=XXXXXXXXXXX`

Reply:
When does `In case there is no status 'cancelled' available yet` apply:?

Reply:
When you return back to the merchant, your payment transaction can have not the correct cancelled status yet. When that happens, it will look in the queryString. Therefore I have implemented the fallback.

Reply:
It feels very unnatural to be sent to a page with the correct status code in the querystring and a request that gets executed later to have an older statuscode. Are you sure this is something that happens?

Reply:
Yes, I know. It does feel very unnatural. It happens in the test and in production environment.